app/src/standard/java/com/infomaniak/mail/firebase/ProcessMessageNotificationsWorker.kt

```diff
@@ -117,11 +122,29 @@ class ProcessMessageNotificationsWorker @AssistedInject constructor(
             SentryLog.i(TAG, "Work finished")
             Result.success()
         }.cancellable().getOrElse {
+            workerFailureReason = it
             Result.failure()
         }.also {
             if (!hasShownNotification) displayGenericNewMailsNotification()
             mailboxContentRealm.close()
         }
+
+        // We don't want to log to Sentry if the error is a common outcome.
+        when (workerFailureReason) {
+            is CancellationException,
+            is NetworkException -> Unit
+            else -> {
+                checkIfNotificationMessageUidIsInRealm(messageUid, mailboxContentRealm)
```

Comment:
The way you wrote it, this will also be checked if the notification is successfully displayed. Is this behavior required?